detector.go

```diff
@@ -101,36 +102,36 @@ func (c *DetectConfig) process(done []Buildpack) ([]Buildpack, []BuildPlanEntry,
 		}
 		run := t.(detectRun)
 		if len(run.Output) > 0 {
-			c.Out.Printf("======== Output: %s ========\n%s", bp, run.Output)
+			c.Logger.Debugf("======== Output: %s ========\n%s", bp, run.Output)
 		}
 		if run.Err != nil {
-			c.Out.Printf("======== Error: %s ========\n%s", bp, run.Err)
+			c.Logger.Debugf("======== Error: %s ========\n%s", bp, run.Err)
 		}
 		runs = append(runs, run)
 	}
 
-	c.Out.Printf("======== Results ========")
+	c.Logger.Debugf("======== Results ========")
 
 	results := detectResults{}
 	detected := true
 	for i, bp := range done {
 		run := runs[i]
 		switch run.Code {
 		case CodeDetectPass:
-			c.Out.Printf("pass: %s", bp)
+			c.Logger.Debugf("pass: %s", bp)
 			results = append(results, detectResult{bp, run})
 		case CodeDetectFail:
 			if bp.Optional {
-				c.Out.Printf("skip: %s", bp)
+				c.Logger.Debugf("skip: %s", bp)
 			} else {
-				c.Out.Printf("fail: %s", bp)
+				c.Logger.Debugf("fail: %s", bp)
```

Comment:
Example of where no new line characters are present yet they should be treated as such when writing out.